packages/calypso-e2e/src/lib/pages/reader-page.ts

```diff
@@ -28,54 +28,42 @@ export class ReaderPage {
 		this.page = page;
 	}
 
-	/**
-	 * Verifies that we are actually on the Reader Page
-	 *
-	 * @returns {Promise<void>} No return value.
-	 */
-	private async verifyReaderPage(): Promise< void > {
-		await this.page.waitForSelector( selectors.readerPageLocator );
-	}
-
 	/**
 	 * Get the URL of the latest post in Reader
 	 *
 	 * @returns {Promise<string>} String of URL for latest post.
 	 */
 	async siteOfLatestPost(): Promise< string > {
-		const href = await this.page.getAttribute( selectors.readerVisitLink, 'href' );
+		const href = await this.page.getAttribute( selectors.visitSiteLink, 'href' );
 		return new URL( href ? href : '' ).host;
 	}
 
 	/**
-	 * Share latest post in Reader
+	 * Visits a post in the Reader.
 	 *
-	 * @returns {Promise<void>} No return value.
+	 * @param param0 Keyed object parameter.
+	 * @param {number} param0.index n-th post to view on the reader page. 1-indexed.
 	 */
-	async shareLatestPost(): Promise< void > {
-		await this.page.click( selectors.shareButtonLocator );
-		await this.page.click( selectors.siteContentLocator );
-	}
+	async visitPost( { index }: { index?: number } = {} ): Promise< void > {
+		let selector = '';
+		if ( index ) {
+			selector = `:nth-match(${ selectors.readerCard }, ${ index })`;
+		}
 
-	/**
-	 * Sets and submits comment on latest post
-	 *
-	 * @param {string} comment Text of the comment.
-	 * @returns {Promise<void>} No return value.
-	 */
-	async commentOnLatestPost( comment: string ): Promise< void > {
-		await this.page.click( selectors.commentButtonLocator );
-		await this.page.fill( selectors.commentTextAreaLocator, comment );
-		await this.page.click( selectors.commentSubmitLocator );
+		await Promise.all( [ this.page.waitForNavigation(), this.page.click( selector ) ] );
 	}
```

Comment:
It looks like `index` is optional is this signature, but it doesn't seem optional in the implementation. If a null index is passed, it will attempt to click an empty selector!

I think that index should be required, and then you can remove the if check too.

Reply:
The intention is to expand this in the future so it can support both selection by index (nth post) or by string text matching. Something similar is already in [`CommentsComponent._click`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/components/comments-component.ts#L70), I just did not want to implement the string matching part for this PR.

Reply:
I thought I had left comments stating this will be stubbed, but apparently I didn't - having that comment would have made my intentions clearer.

The string matching path has been filled out in 9733b7db16d35cbd4f26bf4a17ffa2515f3ca609 and method signature changed slightly to throw if neither index or text params are specified.

Reply:
That makes a lot of sense now, looks good!